main_window/rocket_data.py

```diff
@@ -26,11 +27,15 @@ def __init__(self, device_manager: DeviceManager) -> None:
         self.device_manager = device_manager
 
         self.data_lock = threading.RLock()  # create lock ASAP since self.lock needs to be defined when autosave starts
-        self.timeset: Dict[int, Dict[DataEntryKey, Union[int, float, str]]] = {}
-        self.last_time = 0  # TODO REVIEW/CHANGE THIS, once all subpackets have their own timestamp.
+        # Map: Time -> key-indexed list of data
+        self.timeset: SortedDict[int, Dict[DataEntryKey, Union[int, float, str, Enum]]] = SortedDict()
+        # Map: Key -> time-indexed list of data
+        self.keyset: Dict[DataEntryKey, SortedDict[int, Union[int, float, str, Enum]]] = dict()
+        self.last_time = 0
         self.highest_altitude: Dict[FullAddress, float] = dict()
+        self.existing_entry_keys: Set[DataEntryKey] = set()  # Set of entry keys actually recorded. Used for csv header
```

Comment:
Since now we have the `self.keyset` is there any benefit to keeping `self.existing_entry_keys` vs using `self.keyset.keys`?

Reply:
That's a good point. I guess I implicitly thought it was important for something but didn't realize it was the same and could likely just be removed.